osm2streets/src/output.rs

```diff
@@ -0,0 +1,264 @@
+use itertools::Itertools;
+
+use crate::{Direction, LaneType, StreetNetwork};
+use geo::MapCoordsInPlace;
+use geom::{Distance, Line, PolyLine, Pt2D};
+
+use LaneType::*;
+use SurfaceMaterial::*;
+
+pub struct Surface {
+    pub area: geo::Polygon,
+    pub material: SurfaceMaterial,
+}
+
+pub struct PaintArea {
+    pub area: geo::Polygon,
+    pub color: PaintColor,
+}
+
+impl PaintArea {
+    pub fn new(area: geo::Polygon) -> Self {
+        Self {
+            area,
+            color: PaintColor::White,
+        }
+    }
+    pub fn with_color(area: geo::Polygon, color: PaintColor) -> Self {
+        Self { area, color }
+    }
+}
+
+impl StreetNetwork {
+    /// Generates polygons covering the road, cycle and footpath areas.
+    pub fn get_surfaces(&self) -> Vec<Surface> {
+        let mut output = Vec::new();
+
+        // Add polygons for road surfaces.
+        for road in self.roads.values() {
+            // Generate an area for each contiguous group of footpath, bike path and road lanes.
```

Comment:
I could imagine some uses for this kind of grouping. Are you doing anything with it in JOSM yet, or is it just a bit of a performance saver compared to a polygon per lane?

Reply:
It's an attempt to describe a specific prediction: the paved surfaces described by all the roads and paths. This is an important verification aid for the OSM ethos of describing what's there.
Any drawn curbs and grass and intersection areas would hopefully conflate with our predictions.

An example of a layer that should be topojson?

Reply:
The other slice is designations. Every lane of traffic is a designation, (crossings are flagged somehow) and so is parking and "out of bounds" buffers, gardens and whatnot.

I imagine using lane designation driving lines and hitboxes for mouse interaction. Clicking on a lane to select the appropriate osm way and lane within that for field editing is an attractive idea in JOSM too.

Reply:
It needs to be more sophisticated and recognise that bike lanes on a road are Asphalt, intersections between paths are concrete, etc.

(The "bike lane" designation is how we want to identify the bike lanes themselves. The painted symbols will show up too.)